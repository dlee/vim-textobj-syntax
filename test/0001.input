" Common set up  "{{{1

runtime! plugin/textobj/*.vim

syntax enable




function s:describe__global_variables()  "{{{1
  It "Check global variables"

  Should exists('g:loaded_textobj_syntax') be true
endfunction




function s:describe__named_key_mappings()  "{{{1
  It "Check the existence of the named key mappings in each mode"

  for _ in ['<Plug>(textobj-syntax-a)', '<Plug>(textobj-syntax-i)']
    Should maparg(_, 'c') == ''
    Should maparg(_, 'i') == ''
    Should maparg(_, 'n') == ''
    Should maparg(_, 'o') != ''
    Should maparg(_, 'v') != ''
  endfor
endfunction




function s:describe__select_i()  "{{{1
  It "<Plug>(textobj-syntax-i)"

  enew!
  setfiletype c
  put ='#define ABC 123 /* FIXME: Fix me. */'
  "    0....*....1....*....2....*....3....*.
  "     ------------------------------------  cDefine
  "                 ---                       cNumbers
  "                 ---                       cNumber
  "                     ------------------XX  cComment
  "                     --                --  cCommentStart
  "                        -----              cTodo
  "
  " BUGS: "XX" is not a part of cComment region with the default syntax/c.vim.
  "       But why?  It should be a part of cComment region.

  normal! 1|
  normal yay
  Should @0 ==# getline('.')

  normal! 10|
  normal yay
  Should @0 ==# getline('.')

  normal! 15|
  normal yay
  Should @0 ==# '123'

  normal! 17|
  normal yay
  Should @0 ==# '/*'

  normal! 19|
  normal yay
  Should @0 ==# '/* FIXME: Fix me. '

  normal! 20|
  normal yay
  Should @0 ==# 'FIXME'

  normal! 35|
  normal yay
  Should @0 ==# '*/'
endfunction




function s:describe__select_a()  "{{{1
  " FIXME: Concrete and revise <Plug>(textobj-syntax-a) semantics.
  " FIXME: Currently, the following tests are just a copy of "iy".
  It "<Plug>(textobj-syntax-a)"

  enew!
  setfiletype c
  put ='#define ABC 123 /* FIXME: Fix me. */'
  "    0....*....1....*....2....*....3....*.
  "     ------------------------------------  cDefine
  "                 ---                       cNumbers
  "                 ---                       cNumber
  "                     ------------------XX  cComment
  "                     --                --  cCommentStart
  "                        -----              cTodo
  "
  " BUGS: "XX" is not a part of cComment region with the default syntax/c.vim.
  "       But why?  It should be a part of cComment region.

  normal! 1|
  normal yiy
  Should @0 ==# getline('.')

  normal! 10|
  normal yiy
  Should @0 ==# getline('.')

  normal! 15|
  normal yiy
  Should @0 ==# '123'

  normal! 17|
  normal yiy
  Should @0 ==# '/*'

  normal! 19|
  normal yiy
  Should @0 ==# '/* FIXME: Fix me. '

  normal! 20|
  normal yiy
  Should @0 ==# 'FIXME'

  normal! 35|
  normal yiy
  Should @0 ==# '*/'
endfunction




" __END__  "{{{1
" vim: filetype=vim foldmethod=marker
