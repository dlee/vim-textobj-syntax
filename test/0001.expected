not found in 'runtimepath': "ftdetect/*.vim"
==== global_variables 
---- It "Check global variables" 
.

==== named_key_mappings 
---- It "Check the existence of the named key mappings in each mode" 
..........

==== select_a 
---- It "<Plug>(textobj-syntax-a)" 
.......

==== select_i 
---- It "<Plug>(textobj-syntax-i)" 
.......



**** Result ****


25 examples, 0 failures
